airflow/executors/base_executor.py

```diff
@@ -212,16 +253,20 @@ def trigger_tasks(self, open_slots: int) -> None:
             # removed from the running set in the meantime.
             if key in self.running:
                 attempt = self.attempts[key]
-                if attempt < QUEUEING_ATTEMPTS - 1:
-                    self.attempts[key] = attempt + 1
-                    self.log.info("task %s is still running", key)
+                if attempt.can_try_again():
+                    # if it hasn't been much time since first check, let it be checked again next time
+                    self.log.info("queued but still running; attempt=%s task=%s", attempt.total_tries, key)
                     continue
-
-                # We give up and remove the task from the queue.
-                self.log.error("could not queue task %s (still running after %d attempts)", key, attempt)
-                del self.attempts[key]
-                del self.queued_tasks[key]
+                else:
+                    # Otherwise, we give up and remove the task from the queue.
+                    self.log.error(
+                        "could not queue task %s (still running after %d attempts)", key, attempt.total_tries
+                    )
+                    del self.attempts[key]
+                    del self.queued_tasks[key]
```

Comment:
```suggestion

                # Otherwise, we give up and remove the task from the queue.
                self.log.error("could not queue task %s (still running after %d attempts)", key, attempt.total_tries)
                del self.attempts[key]
                del self.queued_tasks[key]
```

nit: don't need the else

Reply:
here's my take on this...
it's purely a stylistic call.  i think sometimes using the else is positive w.r.t. readability.  and, since there is no automation enforcing one or the other approach, i think we leave it up to individual author.  wdyt?

Reply:
Yeah, absolutely. If you like the else pattern here more, probably worth removing the continue then.

Reply:
mrr.... i like the continue there for readability... i'll just remove the else :)